From 5b570d6a80e83f7675bce29ebffb6f98eb02d30a Mon Sep 17 00:00:00 2001
From: ahs <amrita.h.s@intel.com>
Date: Tue, 23 Jun 2020 16:05:53 +0530
Subject: [PATCH] Make dalvikvm process wait - required for collecting vtune
 profiles

Dalvikvm process is made to wait for creation of wait.txt file in /sdcard
so that we can attach vtune to this process

Change-Id: I019af72613e33c0343ed8ec2c11954321bc86a37
Tracked-On:
Signed-off-by: ahs <amrita.h.s@intel.com>
---
 dalvikvm/dalvikvm.cc | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/dalvikvm/dalvikvm.cc b/dalvikvm/dalvikvm.cc
index e735e2f..ca00e7b 100644
--- a/dalvikvm/dalvikvm.cc
+++ b/dalvikvm/dalvikvm.cc
@@ -20,6 +20,9 @@
 #include <string.h>
 #include <algorithm>
 #include <memory>
+#include <iostream>
+#include <fstream>
+#include <unistd.h>
 
 #include "jni.h"
 #include "nativehelper/JniInvocation.h"
@@ -94,6 +97,17 @@ static int InvokeMain(JNIEnv* env, char** argv) {
     return EXIT_FAILURE;
   }
 
+  std::cout << "dalvikvm waiting for /sdcard/test_data/wait.txt";
+  std::string wait_file_name = "/sdcard/test_data/wait.txt";
+  while(true) {
+    if (FILE *file = fopen(wait_file_name.c_str(), "r")) {
+      fclose(file);
+      std::cout << "AHS:: dalvikvm will resume";
+      break;
+    } else {
+      sleep(1);
+    }
+  }
   // Invoke main().
   env->CallStaticVoidMethod(klass.get(), method, args.get());
 
-- 
2.7.4

